x-pack/plugins/stack_connectors/server/connector_types/security/tines/render.ts

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { set } from 'lodash/fp';
+import { ExecutorParams } from '@kbn/actions-plugin/server/sub_action_framework/types';
+import { RenderParameterTemplates } from '@kbn/actions-plugin/server/types';
+import { SUB_ACTION } from '../../../../common/connector_types/security/tines/constants';
+
+interface Context {
+  alerts: Array<Record<string, unknown>>;
+}
+
+export const renderParameterTemplates: RenderParameterTemplates<ExecutorParams> = (
+  params,
+  variables
+) => {
+  if (params?.subAction !== SUB_ACTION.RUN) return params;
+
+  // Remove the "kibana" entry from all alerts to reduce weight, the same data can be found in other parts of the alert object.
+  const alerts = (variables.context as Context).alerts.map(({ kibana, ...alert }) => alert);
```

Comment:
In Kibana we can always migrate SO when changing the schema. I am not sure what changes the alert-as-data initiative will do so I just wanted to raise awareness of a pontentionally issue. Maybe I am wrong and we do not need to do anything. @ymao1 Any input on this?

Reply:
Thanks for bringing this up @cnasikas!

This is an accurate assessment as right now, each rule type defines its own context variables that it populates to pass through to actions/notifications. If a rule type has `context.variableA` and changes it to `context.variableB`, we don't have a process for migrating the mustache templates that are stored in the rule saved object and the actions that use `context.variableA` would render empty data.

That said, this is assuming a very detection rule specific context. Maybe that's ok with this PR since it'll just be enabled for detection rules currently but I would like to see some error handling for the case where `context.alerts` is not defined (as is the case for every rule type other than detection rules).

Also, I would agree with @cnasikas that it an allowlist makes sense. At a minimum, I would remove the `kibana.alert.rule.*` keys since we are evaluating the necessity of keeping them in AAD and the information in them is stored elsewhere (in `params` and at the top level `rule` object).

Reply:
Oh I see! That is even better!